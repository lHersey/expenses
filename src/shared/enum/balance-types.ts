export enum BalanceType {
  INCOME,
  EXPENSE,
}
