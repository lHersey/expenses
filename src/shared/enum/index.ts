export * from './balance-types';
export * from './transaction-types';
