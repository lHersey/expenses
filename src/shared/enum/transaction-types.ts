export enum TransactionType {
  INCOME,
  EXPENSE,
  TRANSFER,
}
