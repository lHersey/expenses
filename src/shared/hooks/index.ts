export * from './useColorAdjust';
export * from './useModalState';
export * from './useTabScreenOptions';
