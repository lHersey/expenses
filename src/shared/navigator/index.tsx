import { NavigationContainer } from '@react-navigation/native';
import React, { FC } from 'react';

import MainTabNavigator from './main-tab-navigator';

const Navigator: FC = () => {
  return (
    <NavigationContainer>
      <MainTabNavigator />
    </NavigationContainer>
  );
};

export default Navigator;
