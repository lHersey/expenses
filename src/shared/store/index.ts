import AsyncStorage from '@react-native-async-storage/async-storage';
import { configureStore } from '@reduxjs/toolkit';
import { persistReducer, persistStore } from 'redux-persist';

import rootReducer from './reducers';

const persistedReducer = persistReducer(
  {
    key: 'PARKING',
    storage: AsyncStorage,
  },
  rootReducer,
);

const store = configureStore({
  reducer: persistedReducer,
  devTools: __DEV__,
  middleware: getDefaultMiddleware => getDefaultMiddleware({ serializableCheck: false }),
});

const persistor = persistStore(store);

export { persistor };
export default store;
