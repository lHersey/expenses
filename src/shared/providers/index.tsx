import React, { FC } from 'react';
import FlashMessage from 'react-native-flash-message';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { Provider as StoreProvider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import store, { persistor } from 'shared/store';
import { ThemeProvider } from 'styled-components/native';

import { THEME } from '../../styles/theme';

const Providers: FC = props => {
  const { children } = props;

  return (
    <StoreProvider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <ThemeProvider theme={THEME}>
          <SafeAreaProvider>
            <FlashMessage position="center" />
            {children}
          </SafeAreaProvider>
        </ThemeProvider>
      </PersistGate>
    </StoreProvider>
  );
};

export default Providers;
