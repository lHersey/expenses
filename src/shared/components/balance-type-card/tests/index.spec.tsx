import { render } from '@testing-library/react-native';
import React from 'react';
import { BalanceType } from 'shared/enum';
import Providers from 'shared/providers/test-providers';

import BalanceTypeCard from '../index';

describe('<BalanceTypeCard />', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
      expect.hasAssertions();
      const rendered = render(
        <Providers>
          <BalanceTypeCard ammount={4000} type={BalanceType.EXPENSE} />
        </Providers>,
      ).toJSON();

      expect(rendered).toMatchSnapshot();
    });
  });

  describe('unit', () => {
    it('should render the text', () => {
      expect.hasAssertions();
      const { queryByText } = render(
        <Providers>
          <BalanceTypeCard ammount={4000} type={BalanceType.EXPENSE} />
        </Providers>,
      );

      expect(queryByText('$4000.00')).toBeTruthy();
    });
  });
});
