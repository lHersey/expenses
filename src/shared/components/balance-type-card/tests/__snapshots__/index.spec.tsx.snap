// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BalanceTypeCard /> snapshots should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.15)",
          "borderRadius": 10,
          "display": "flex",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "marginBottom": 10,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  >
    <Text />
    <View
      style={
        Array [
          Object {
            "marginLeft": 10,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#5e5e5e",
              "fontFamily": "Inter_500",
              "fontSize": 16,
            },
            Object {
              "color": "#e3e3e3",
              "fontFamily": "Inter_700",
              "fontSize": 14,
            },
          ]
        }
      >
        Expenses
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#5e5e5e",
              "fontFamily": "Inter_500",
              "fontSize": 16,
            },
            Object {
              "color": "#e3e3e3",
              "fontSize": 14,
            },
          ]
        }
      >
        $
        4000
        .00
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
